webcrypto/aes.go

```diff
@@ -151,3 +155,458 @@ func importAESKey(
+	// The counter must be big enough that it doesn't wrap: if the message is n blocks and the counter is m bits long, then
+	// the following must be true: n <= 2^m.
+	//
+	// The NIST SP800-38A standard, which defines CTR, suggests that the counter should occupy half of the counter
+	// block (see Appendix B.2), so for AES it would be 64.
+	Length int `json:"length"`
+}
+
+// Encrypt encrypts the given plaintext using the AES-CTR algorithm, and returns the ciphertext.
+// Implements the WebCryptoAPI's `encrypt` method's [specification] for the AES-CTR algorithm.
+//
+// [specification]: https://www.w3.org/TR/WebCryptoAPI/#aes-ctr
+func (acp *AesCtrParams) Encrypt(plaintext []byte, key CryptoKey) ([]byte, error) {
+	// 1.
+	// Note that aes.BlockSize stands for the `k` variable as per the specification.
+	if len(acp.Counter) != aes.BlockSize {
+		return nil, NewError(0, OperationError, "counter length is not 16 bytes")
+	}
+
+	// 2.
+	if acp.Length == 0 || acp.Length > 128 {
+		return nil, NewError(0, OperationError, "invalid counter length, out of the 0 < x < 128 bounds")
+	}
+
+	keyHandle, ok := key.handle.([]byte)
+	if !ok {
+		return nil, NewError(0, ImplementationError, "could not get key handle")
+	}
+
+	// 3.
+	block, err := aes.NewCipher(keyHandle)
+	if err != nil {
+		return nil, NewError(0, OperationError, "could not create cipher")
+	}
+
+	ciphertext := make([]byte, len(plaintext))
+	ctr := cipher.NewCTR(block, acp.Counter)
+	ctr.XORKeyStream(ciphertext, plaintext)
+
+	return ciphertext, nil
+}
+
+// Decrypt decrypts the given ciphertext using the AES-CTR algorithm, and returns the plaintext.
+// Implements the WebCryptoAPI's `decrypt` method's [specification] for the AES-CTR algorithm.
+//
+// [specification]: https://www.w3.org/TR/WebCryptoAPI/#aes-ctr
+func (acp *AesCtrParams) Decrypt(ciphertext []byte, key CryptoKey) ([]byte, error) {
+	// 1.
+	if len(acp.Counter) != aes.BlockSize {
+		return nil, NewError(0, OperationError, "counter length is invalid, should be 16 bytes")
+	}
+
+	// 2.
+	if acp.Length == 0 || acp.Length > 128 {
+		return nil, NewError(0, OperationError, "invalid length, should be within 1 <= length <= 128 bounds")
+	}
+
+	keyHandle, ok := key.handle.([]byte)
+	if !ok {
+		return nil, NewError(0, OperationError, "invalid key handle")
+	}
+
+	// 3.
+	block, err := aes.NewCipher(keyHandle)
+	if err != nil {
+		return nil, NewError(0, OperationError, "could not create AES cipher")
+	}
+
+	plaintext := make([]byte, len(ciphertext))
+	stream := cipher.NewCTR(block, acp.Counter)
+	stream.XORKeyStream(plaintext, ciphertext)
+
+	return plaintext, nil
+}
+
+var (
+	// Ensure that AesCtrParams implements the Encrypter interface.
+	_ Encrypter = &AesCtrParams{}
+
+	// Ensure that AesCtrParams implements the Decrypter interface.
+	_ Decrypter = &AesCtrParams{}
+
+	// Ensure that AesCtrParams implements the EncryptDecrypter interface.
+	_ EncryptDecrypter = &AesCtrParams{}
```

Comment:
```suggestion
	// Ensure that AesCtrParams implements the EncryptDecrypter interface.
	_ EncryptDecrypter = &AesCtrParams{}
```
EncryptDecrypter should be enough for all of them